### PR TITLE
WINC-645: [controllers] Handle generic events

### PR DESCRIPTION
**Pull request by aravindhp**
- Simplify the check for a Windows node
- Update all controllers to handle generic events

### CODE REVIEW

File: controllers/controllers.go [hunk 1]
```diff
@@ -187,3 +184,15 @@ func isValidWindowsNode(o client.Object, byoh bool) bool {
...
+// isWindowsNodeWithVersionMismatch returns true if the node object is a valid Windows node and if there is a
+// version mismatch which will require the node to be updated.
+func isWindowsNodeWithVersionMismatch(o client.Object, byoh bool) bool {
+	if !isValidWindowsNode(o, byoh) {
+		return false
+	}
+	if o.GetAnnotations()[metadata.VersionAnnotation] != version.Get() {
+		return true
+	}
+	return false
```
Comment (saifshaikh48) [inline controllers/controllers.go:197]: Can this function be reduced to `return isValidWindowsNode(o, byoh) && o.GetAnnotations()[metadata.VersionAnnotation] != version.Get()`? And if so, is there a need to keep this in a helper? I am fine either way, just floating the idea.

**Review by saifshaikh48 — COMMENTED** [review]
Thanks for taking this on @aravindhp, PTAL at my comment.

File: controllers/controllers.go (hunk 1 above)
Comment (aravindhp) [reply controllers/controllers.go:197]: I am fine removing the helper.

Comment (mansikulkarni96) [reply controllers/controllers.go:197]: +1 not sure we need another helper, rest LGTM.

Comment (aravindhp) [reply controllers/controllers.go]: Addressed. PTAL.

Comment (mansikulkarni96) [reply controllers/controllers.go]: The PR description needs an update removing first bullet point.

Comment (aravindhp) [reply controllers/controllers.go]: Addressed. PTAL.

**Review by jrvaldes — APPROVED** [review]
/lgtm

File: controllers/controllers.go (hunk 1 above)
Comment (saifshaikh48) [reply controllers/controllers.go]: /lgtm

Comment (mansikulkarni96) [reply controllers/controllers.go]: LGTM

**Pull request merged**